mne/time_frequency/tfr.py

```diff
@@ -2860,6 +2860,41 @@ def average(self, method="mean", dim="epochs", copy=False):
             self.freqs = freqs
             return self
 
+    @verbose
+    def drop(self, indices, reason="USER", verbose=None):
+        """Drop epochs based on indices or boolean mask.
+
+        .. note:: The indices refer to the current set of undropped epochs
+                  rather than the complete set of dropped and undropped epochs.
+                  They are therefore not necessarily consistent with any
+                  external indices (e.g., behavioral logs). To drop epochs
+                  based on external criteria, do not use the ``preload=True``
+                  flag when constructing an Epochs object, and call this
+                  method before calling the :meth:`mne.Epochs.drop_bad` or
+                  :meth:`mne.Epochs.load_data` methods.
+
+        Parameters
+        ----------
+        indices : array of int or bool
+            Set epochs to remove by specifying indices to remove or a boolean
+            mask to apply (where True values get removed). Events are
+            correspondingly modified.
+        reason : str
+            Reason for dropping the epochs ('ECG', 'timeout', 'blink' etc).
+            Default: 'USER'.
+        %(verbose)s
+
+        Returns
+        -------
+        epochs : instance of Epochs
+            The epochs with indices dropped. Operates in-place.
+        """
+        from ..epochs import BaseEpochs
+
+        BaseEpochs.drop(self, indices=indices, reason=reason, verbose=verbose)
```

Comment:
@larsoner @cbrnr this is the bit we specifically wanted your input on: here @CarinaFo is defining `EpochsTFR.drop()` as a wrapper for `BaseEpochs.drop()`.  Are you OK with this approach, or prefer moving `BaseEpochs.drop()` to a mixin class that both `BaseEpochs` and `EpochsTFR` inherit from?

Reply:
So I assume the wrapper is okey? I would close the PR then?

Reply:
In principle I think we usually use a mixin class for this but I think it's okay to do it this way for now